packages/block-editor/src/hooks/position.js

```diff
@@ -0,0 +1,201 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Platform } from '@wordpress/element';
+import { getBlockSupport } from '@wordpress/blocks';
+import {
+	__experimentalToggleGroupControl as ToggleGroupControl,
+	__experimentalToggleGroupControlOption as ToggleGroupControlOption,
+} from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import useSetting from '../components/use-setting';
+import { LAYOUT_SUPPORT_KEY } from './layout';
+import { cleanEmptyObject } from './utils';
+
+const POSITION_OPTIONS = [
+	{
+		key: 'default',
+		label: __( 'Default' ),
+		value: '',
+		name: __( 'Default' ),
+	},
+	{
+		key: 'sticky',
+		label: __( 'Sticky' ),
+		value: 'sticky',
+		name: __( 'Sticky' ),
+	},
+];
+
+const POSITION_SIDES = [ 'top', 'right', 'bottom', 'left' ];
+const VALID_POSITION_TYPES = [ 'sticky', 'fixed' ];
+
+/**
+ * Get calculated position CSS.
+ *
+ * @param {Object} props          Component props.
+ * @param {string} props.selector Selector to use.
+ * @param {Object} props.style    Style object.
+ * @return {string} The generated CSS rules.
+ */
+export function getPositionCSS( { selector, style } ) {
+	let output = '';
+
+	const { position } = style?.layout || {};
+
+	if ( ! VALID_POSITION_TYPES.includes( position ) ) {
+		return output;
+	}
+
+	output += `${ selector } {`;
+	output += `position: ${ position };`;
+
+	POSITION_SIDES.forEach( ( side ) => {
+		if ( style?.layout?.[ side ] !== undefined ) {
+			output += `${ side }: ${ style.layout[ side ] };`;
+		}
+	} );
+
+	if ( position === 'sticky' || position === 'fixed' ) {
+		// TODO: Work out where to put the magic z-index value.
+		output += `z-index: 250`;
```

Comment:
Could the z-index be defined in theme.json `layout.definitions`, e.g., `is-layout-sticky`  similar to `is-layout-flow`?

🤷

Reply:
Just jotting down a note here that it might make sense to have some preset-like values for `z-index` stored in `settings.layout.zIndexSticky` or if we want support for multiple z-indexes, then perhaps an array of z-indexes for different purposes. That way we could have a core default z-index value for sticky / fixed position, but themes could then override it with their own values if need be.